File: Leetcode/cpp/sol1800.cpp

// https://leetcode.com/problems/maximum-ascending-subarray-sum/description/

class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        if(nums.size() == 1){
            return nums[0];
        }

        int max_sum = nums[0];
        int temp = nums[0];
        int curr = 1;
        while (curr < nums.size()){
            if(nums[curr-1] < nums[curr]){
                temp+=nums[curr];
            }else{
                max_sum = max(max_sum, temp);
                temp = nums[curr];
            }
            curr++;
        }

        return max(temp, max_sum);
    }
};

File: Leetcode/python/sol1800.py

# https://leetcode.com/problems/maximum-ascending-subarray-sum/description/
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        if len(nums) == 1: # constraints specify that len(nums) is between 1 to 100
            return nums[0]
        start = 0
        temp = 0
        max_sum = float('-inf')
        while start < len(nums)-1:
            if nums[start] < nums[start+1]:
                temp+=nums[start]
            else:
                temp+=nums[start]
                max_sum = max(max_sum, temp)
                temp = 0
            start+=1
        else:
            if nums[start-1] < nums[start]:
                temp+=nums[start]
                max_sum = max(max_sum, temp)

        return max_sum